features/steps/test_behave.py

```diff
@@ -0,0 +1,244 @@
+#!/usr/bin/python -tt
+
+from behave import *
+import os
+import subprocess
+import glob
+import re
+import shutil
+
+DNF_FLAGS = ['-y', '--disablerepo=*', '--nogpgcheck']
+RPM_INSTALL_FLAGS = ['-Uvh']
+RPM_ERASE_FLAGS = ['-e']
+
+
+def _left_decorator(item):
+    """ Removed packages """
+    return u'-' + item
+
+
+def _right_decorator(item):
+    """ Installed packages """
+    return u'+' + item
+
+
+def find_pkg(pkg):
+    """ Find the package file in the repository """
+    candidates = glob.glob('/repo/' + pkg + '*.rpm')
+    if len(candidates) == 0:
+        print("No candidates for: '{0}'".format(pkg))
+    assert len(candidates) == 1
+    return candidates[0]
+
+
+def decorate_rpm_packages(pkgs):
+    """ Converts package names like TestA, TestB into absolute paths """
+    return [find_pkg(p) for p in pkgs]
+
+
+def get_rpm_package_list():
+    """ Gets all installed packages in the system """
+    pkgstr = subprocess.check_output(['rpm', '-qa', '--queryformat', '%{NAME}\n'])
+    return pkgstr.splitlines()
+
+
+def get_rpm_package_version_list():
+    """ Gets all installed packages in the system with version"""
+    pkgverstr = subprocess.check_output(['rpm', '-qa', '--queryformat', '%{NAME}-%{VERSION}-%{RELEASE}\n'])
+    return pkgverstr.splitlines()
+
+
+def get_dnf_package_version_list():
+    """ Gets all installed packages in the system with version to check that dnf has same data about installed packages"""
+    pkgverstr = subprocess.check_output(['dnf', 'repoquery', '--installed', '-Cq', '--queryformat', '%{name}.%{version}.%{release}\n'])
+    pkgverstr = pkgverstr.splitlines()
+    return pkgverstr
+
+
+def diff_package_lists(a, b):
+    """ Computes both left/right diff between lists `a` and `b` """
+    sa, sb = set(a), set(b)
+    return (map(_left_decorator, list(sa - sb)),
+        map(_right_decorator, list(sb - sa)))
+
+
+def package_version_lists(pkg, list_ver):
+    """ Select package versions """
+    found_pkgs = [x for x in list_ver if re.search('^' + pkg, x)]
+    assert len(found_pkgs) == 1
+    return str(found_pkgs[0])
+
+
+def package_absence(pkg, list_ver):
+    """ Select package versions """
+    found_pkgs = [x for x in list_ver if re.search('^' + pkg, x)]
```

Comment:
`... if x.startswith(pkg)...` (you know for the next time, regexps are slower than string comparison)